File: algorithms/2006.count-number-of-pairs-with-absolute-difference-k/solution.go
func countKDifference(nums []int, k int) int {
	count := 0

	for i := 0; i < len(nums)-1; i++ {
		for j := i + 1; j < len(nums); j++ {
			if int(math.Abs(float64(nums[i])-float64(nums[j]))) == k {
				count++
			}
		}
	}

	return count
}

File: algorithms/2006.count-number-of-pairs-with-absolute-difference-k/solution.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer}
def count_k_difference(nums, k)
  count = 0

  (0...nums.size - 1).each do |i|
    (i + 1...nums.size).each do |j|
      count += 1 if (nums[i] - nums[j]).abs == k
    end
  end

  count
end
